core/DataAccess/RawLogDao.php

```diff
@@ -103,23 +103,39 @@ public function countVisitsWithDatesLimit($from, $to)
      *                            ```
      * @param int $iterationStep The number of rows to query at a time.
      * @param callable $callback The callback that processes each chunk of rows.
+     * @param string $willDelete Set to true if you will make sure to delete all rows that were fetched. If you are in
+     *                           doubt and not sure if to set true or false, use "false". Setting it to true will
+     *                           enable an internal performance improvement but it can result in an endless loop if not
+     *                           used properly.
      */
-    public function forAllLogs($logTable, $fields, $conditions, $iterationStep, $callback, $useReader = false)
+    public function forAllLogs($logTable, $fields, $conditions, $iterationStep, $callback, $willDelete)
     {
-        $idField = $this->getIdFieldForLogTable($logTable);
-        list($query, $bind) = $this->createLogIterationQuery($logTable, $idField, $fields, $conditions, $iterationStep);
-
         $lastId = 0;
-        if ($useReader) {
-            $db = Db::getReader();
```

Comment:
we can no longer use the reader for this but that's fine as the query is now fast.

If we were still using the reader, we would risk reading visits again that were already deleted on the master but the delete was not yet replicated to the reader...